src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -597,6 +597,9 @@ impl<'a, 'tcx> BorrowckCtxt<'a, 'tcx> {
                         if let Ok(ty::ClosureKind::FnOnce) =
                            ty::queries::closure_kind::try_get(self.tcx, DUMMY_SP, id) {
                             err.help("closure was moved because it only implements `FnOnce`");
+                            if let Some(&(_kind, Some(span))) = self.tables.closure_kinds.get( ) {
```

Comment:
This would be the result of `self.tcx.hir.as_local_node_id(id)`, I think -- we're basically looking things up based on the id of the closure. However, what we might want to do is just to include this information in the result of `closure_kind::try_get()` above, or else make a separate query.